addons/mail/models/res_users.py

```diff
@@ -111,6 +119,31 @@ def write(self, vals):
                         subtype_xmlid='mail.mt_note'
                     )
 
+        if 'login' in vals:
+            self._notify_security_setting_update(
+                _('Security Update: Login Changed'),
+                _("This email to let you know that your account login has been updated."),
+            )
+        if 'password' in vals:
+            self._notify_security_setting_update(
+                _('Security Update: Password Changed'),
+                _("This email to let you know that your account password has been updated."),
+            )
+        if 'email' in vals:
+            # when the email is modified, we want notify the previous address (and not the new one)
+            for user, previous_email in previous_email_by_user.items():
+                if previous_email:
```

Comment:
Maybe filter upfront so `previous_email_by_user` only has entries for users which had an email?